src/features/party/partyProviderUtils.ts

```diff
@@ -0,0 +1,40 @@
+import { type IParty, PartyType } from 'src/types/shared';
+import type { IApplicationMetadata } from 'src/features/applicationMetadata';
+
+export const flattenParties = (parties: IParty[]): IParty[] => {
+  const result: IParty[] = [];
+  const stack = [...parties];
+
+  while (stack.length) {
+    const current = stack.pop();
+    if (current) {
+      result.push(current);
+      if (current.childParties) {
+        stack.push(...current.childParties);
+      }
+    }
+  }
+
+  return result;
+};
+
+export const GetValidParties = (parties: IParty[], appMetadata: IApplicationMetadata): IParty[] => {
```

Comment:
Since you wanted nitpicks - I think we usually reserve PascalCase for classes, and camelCase for functions (so since this is a function, i would have called it `getValidParties` or `reduceToValidParties`).